lamindb/models/query_set.py

```diff
@@ -523,6 +523,21 @@ class BasicQuerySet(models.QuerySet):
         >>> queryset
     """
 
+    def __new__(cls, model=None, query=None, using=None, hints=None):
+        from lamindb.models import Artifact, ArtifactSet
+
+        # If the model is Artifact, create a new class
+        # for BasicQuerySet or QuerySet that inherits from ArtifactSet.
+        # This allows to add artifact specific functionality to all classes
+        # inheriting from BasicQuerySet.
+        # Thus all query sets of artifacts (and only of artifacts)
+        # will have functions from ArtifactSet.
+        if model is Artifact and not issubclass(cls, ArtifactSet):
+            new_cls = type("Artifact" + cls.__name__, (cls, ArtifactSet), {})
```

Comment:
It looks like ArtifactSet no longer behaves like a QuerySet, does it? It e.g. lacks `.filter()`.

Can we no longer chain ArtifactSet then?

```
ln.Artifact.filter(ulabels__name="experiment 1").filter(ulabels__name="DMSO")
```

Reply:
no, no, it works.